test/lit/passes/remove-unused-module-elements-closed-world.wast

```diff
@@ -0,0 +1,32 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: wasm-opt %s --remove-unused-module-elements --closed-world -all -S -o - | filecheck %s
+
+(module
+  ;; CHECK:      (type $none_=>_none (func))
+
+  ;; CHECK:      (elem declare func $unused)
+
+  ;; CHECK:      (export "foo" (func $exported))
+
+  ;; CHECK:      (func $unused (type $none_=>_none)
+  ;; CHECK-NEXT:  (unreachable)
+  ;; CHECK-NEXT: )
+  (func $unused
+    ;; Even though we take a reference to this function, we do not call it and
+    ;; we are assuming the outside world won't call it, so we can optimize out
+    ;; its body.
+    (nop)
+  )
+
+  ;; CHECK:      (func $exported (type $none_=>_none)
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (ref.func $unused)
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $exported (export "foo")
+    ;; This pass can't prove that this reference won't escape.
+    (drop
+      (ref.func $unused)
+    )
+  )
+)
```

Comment:
Is this single test enough? The old file had several places that checked the ability of closed-world to optimize things like this. I think it had various corner cases there.

Reply:
I think this test is enough because the only place we use `closedWorld` in RemoveUnusedModuleElements is to determine what to do in precisely this situation.

Reply:
Skimming it now, I see we check various orderings of how `RefFunc`s appear interleaved with `CallRef`s. We also have things like `Verify that we do not crash on an unreachable call_ref` that I think is worth checking in both modes. So I do think we want to basically convert this file to work in closed world rather than add a single tailored test. If you want I can look into that.

Reply:
Ah, right, because all these tests predate support for _open_ world, not closed world. I can fix it up.